Packs/FeedCyrenThreatInDepth/Integrations/CyrenThreatInDepth/CyrenThreatInDepth_test.py

```diff
@@ -0,0 +1,576 @@
+
+
+@pytest.fixture(name="phishing_urls", scope="session")
+def fixture_phishing_urls():
+    return _load_file("phishing_urls.jsonl")
+
+
+@pytest.fixture(name="ip_reputation", scope="session")
+def fixture_ip_reputation():
+    return _load_file("ip_reputation.jsonl")
+
+
+@pytest.fixture(name="response_429", scope="session")
+def fixture_response_429():
+    return _load_file("429.html")
+
+
+@pytest.fixture(name="clean_integration_context", scope="function")
+def fixture_clean_integration_context():
+    demisto.setIntegrationContext({})
+    yield
+    demisto.setIntegrationContext({})
+
+
+def _create_instance(requests_mock, feed, feed_data, offset_data, offset=0, count=2):
+    base_url = "https://cyren.feed/"
+    requests_mock.get(base_url + "data?format=jsonl&feedId={}&offset={}&count={}".format(feed, offset, count),
+                      text=feed_data)
+    requests_mock.get(base_url + "info?format=jsonl&feedId={}".format(feed), json=offset_data)
+    client = Client(feed_name=feed, base_url=base_url, verify=False, proxy=False)
+
+    def fetch_command(initial_count=0, max_indicators=2, update_context=False):
+        return fetch_indicators_command(client, initial_count, max_indicators, update_context)
+
+    def get_command(max_indicators):
+        args = dict(max_indicators=max_indicators)
+        return get_indicators_command(client, args)
+
+    return fetch_command, get_command
+
+
+@pytest.mark.parametrize(
+    "context_data, offsets, initial_count, max_indicators, expected_offset, expected_count", [
+        # first run, no previous offset stored, don't want to have initial
+        # import, trying to get 2 in total
+        (dict(), dict(startOffset=1, endOffset=10000), 0, 2, 10000, 2),
+        # first run, no previous offset stored, want 1000 as initial
+        # import, trying to get 1002 in total
+        (dict(), dict(startOffset=1, endOffset=10000), 1000, 2, 9001, 1002),
+        # not the first run, next accepted offset is 9001, initial import
+        # ignored, getting just the maximum of 2
+        (dict(offset=9001), dict(startOffset=1, endOffset=10000), 0, 2, 9001, 2),
+        (dict(offset=9001), dict(startOffset=1, endOffset=10000), 10000, 2, 9001, 2),
+        # not the first run, next accepted offset is 9001, initial import
+        # ignored, getting a maximum of 2000, even though more would be
+        # available
+        (dict(offset=9001), dict(startOffset=1, endOffset=10000), 0, 2000, 9001, 2000),
+        (dict(offset=9001), dict(startOffset=1, endOffset=10000), 10000, 2000, 9001, 2000),
+        # count is capped at 100000
+        (dict(), dict(startOffset=1, endOffset=10000), 0, 100001, 10000, 100000),
+        (dict(offset=9001), dict(startOffset=1, endOffset=10000), 0, 100001, 9001, 100000),
+    ]
+)
+def test_fetch_indicators_offsets(requests_mock, ip_reputation, context_data, offsets,
+                                  initial_count, max_indicators, expected_offset, expected_count):
```

Comment:
Unit tests should contain given-when-then comment to describe the unit test scenario. `Given <some input>, When <some state> Then <ensure some behviour>`.

Please add those to all unit tests in this file.
See the `iDefense` pack `iDefense_v2_test.py` unit tests for a reference.